src/server.rs

```diff
@@ -474,20 +481,22 @@ async fn handle_push_request(
                 .body(Body::empty())
                 .unwrap())
         }
-        Err(e) => {
-            warn!("Error: {}", e);
-            Ok(Response::builder()
-                .status(match e {
-                    SendPushError::SendError(_) => StatusCode::BAD_GATEWAY,
-                    SendPushError::ProcessingClientError(_) => StatusCode::BAD_REQUEST,
-                    SendPushError::ProcessingRemoteError(_) => StatusCode::BAD_GATEWAY,
-                    SendPushError::AuthError(_) => StatusCode::INTERNAL_SERVER_ERROR,
-                    SendPushError::Other(_) => StatusCode::INTERNAL_SERVER_ERROR,
-                })
-                .header(CONTENT_TYPE, "text/plain")
-                .body(Body::from("Push not successful"))
-                .unwrap())
-        }
+        Err(e) => Ok(Response::builder()
+            .status({
+                info!("{e}");
+                match e {
+                    SendPushError::RemoteServer(_) | SendPushError::RemoteAuth(_) => {
+                        StatusCode::BAD_GATEWAY
+                    }
```

Comment:
Is the `RemoteAuth` error something that depends on the credentials passed in by the user, or only on the server config file? In both cases "bad gateway" might be the wrong status code.

If it depends on the server config, then retrying might help (in case the server config was fixed), but it's not the gateway's (i.e. push service's) problem. In that case, "internal server error" (as before) might be better.

If it depends on the user's credentials, then "bad request" would be better.

Reply:
True ... Changed to internal server error